examples/src/App.js

```diff
@@ -1,226 +1,55 @@
 import React, { useState } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { v4 as uuidv4 } from 'uuid';
 import {
   SwipeableList,
   SwipeableListItem
 } from '@sandstreamdev/react-swipeable-list';
 import '@sandstreamdev/react-swipeable-list/dist/styles.css';
 
-import ListItem from './ComplexListItem';
-import ComplexItemContent from './ComplexItemContent';
-import { MailIcon, ReplyIcon, DeleteIcon } from '../images/icons';
 import styles from './app.module.css';
+import transitionStyles from './transitions.module.css';
 
 function App() {
-  const [triggeredSimpleItemAction, triggerSimpleItemAction] = useState('None');
-  const [triggeredComplexItemAction, triggerComplexItemAction] = useState('');
-  const [swipeProgress, handleSwipeProgress] = useState();
-  const [swipeAction, handleSwipeAction] = useState('None');
-
-  const swipeRightDataSimple = name => ({
-    content: (
-      <div className={styles.contentLeft}>
-        <span>Left content</span>
-      </div>
-    ),
-    action: () => triggerSimpleItemAction(`Swipe right action on "${name}"`)
-  });
-
-  const swipeLeftDataSimple = name => ({
-    content: (
-      <div className={styles.contentRight}>
-        <span>Right content</span>
-      </div>
-    ),
-    action: () => triggerSimpleItemAction(`Swipe left action on "${name}"`)
-  });
-
-  const itemContentSimple = name => (
-    <div className={styles.listItem}>
-      <span>{name}</span>
-    </div>
-  );
-
-  const swipeRightDataComplex = name => ({
-    content: (
-      <ComplexItemContent
-        icon={<DeleteIcon />}
-        label="Delete"
-        side="right"
-        color="red"
-      />
-    ),
-    action: () =>
-      triggerComplexItemAction(`Delete action triggered on "${name}" item`)
-  });
-
-  const swipeLeftDataComplex = name => ({
-    content: (
-      <ComplexItemContent
-        icon={<ReplyIcon />}
-        label="Reply"
-        color="green"
-        side="left"
-      />
-    ),
-    action: () =>
-      triggerComplexItemAction(`Reply action triggered on "${name}" item`)
-  });
-
-  const handleSwipeStart = () => {
-    triggerSimpleItemAction('None');
-    handleSwipeAction('Swipe started');
-  };
-
-  const handleSwipeEnd = () => {
-    handleSwipeAction('Swipe ended');
-    handleSwipeProgress();
-  };
+  const [items, setItems] = useState([
```

Comment:
The initial state should be lifted out of scope as at every render the found `uuidv4` calls would fire causing side-effects. If you wish to not lift it up, please wrap it inside function so the `useState` hook will lazily call it only once:

```suggestion
  const [items, setItems] = useState(() => [...])
```